lib/stream/http.go

```diff
@@ -53,7 +56,25 @@ func (h *HTTP) Produce() (io.Reader, io.Reader, error) {
 		return nil, nil, err
 	}
 
-	return h.Res.Body, nil, nil
+	if h.Res.StatusCode == 200 {
+		return h.Res.Body, nil, nil
+	}
+
+	// Try to glean some additional information by reading from the response
+	// body.
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(h.Res.Body)
```

Comment:
I think you still have to close the Body

Reply:
This is done in line 80, but I guess I should just do it here?